master/security/kubernetes-default-deny.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Enable default deny for Kubernetes pods
+---
+
+### Big picture
+
+Change the default for Kubernetes pods from allowing all traffic, to denying all traffic using Kubernetes or Calico network policy.  
+
+### Value
+
+A **default deny** network policy provides an enhanced security posture -- so workloads without policy (or incorrect policy) are not allowed traffic until appropriate network policy is defined.
+
+### Features
+
+This how-to guide uses the following Calico features:
+- **NetworkPolicy** 
+- **GlobalNetworkPolicy**
+
+### Concepts
+
+#### Default deny/allow behavior
+
+**Default allow** means all traffic is allowed by default, unless otherwise specified. **Default deny** means all traffic is denied by default, unless explicitly allowed. **Kubernetes pods are default allow**, unless network policy is defined to specify otherwise.
+
+For compatibility with Kubernetes, **Calico network policy** enforcement follows the standard convention for Kubernetes pods:
+- If no network policies apply to a pod, then all traffic to/from that pod is allowed.
+- If one or more network policies apply to a pod with type ingress, then only the ingress traffic specifically allowed by those policies is allowed, and all other traffic to/from the pod is denied.
+- If one or more network policies apply to a pod with type egress, then only the egress traffic specifically allowed by those policies is allowed, and all other traffic to/from the pod is denied.
+
+For other endpoint types (VMs, host interfaces), **Calico global network policy is default deny**. Only traffic specifically allowed by network policy is allowed, even if no network policies apply to the endpoint.
```

Comment:
> For other endpoint types (VMs, host interfaces), **Calico global network policy is default deny**.

I think this should say "For other endpoint types (VMs, host interfaces), **the default behavior is to deny traffic**. 

The bit that I find off about the current wording is that it implies Calico global network policy objects have something to do with this, when in fact this is true with or without any GNPs specified.

Reply:
good point. Fixed.